openjdk.test.load/src/test.load/net/adoptopenjdk/stf/MixedLoadTest.java

```diff
@@ -22,15 +22,34 @@
 import net.adoptopenjdk.stf.processes.definitions.JavaProcessDefinition;
 import net.adoptopenjdk.stf.processes.definitions.LoadTestProcessDefinition;
 import net.adoptopenjdk.stf.runner.modes.HelpTextGenerator;
+import net.adoptopenjdk.stf.util.TimeParser;
 import net.adoptopenjdk.loadTest.InventoryData;
+import net.adoptopenjdk.stf.environment.StfTestArguments; 
 
 public class MixedLoadTest implements StfPluginInterface {
+	
+	String timeLimit;
+	int testCountMultiplier;
+	boolean isTimeBasedLoadTest = true;
+	String defaultTimeout = "3h";
+	
 	public void help(HelpTextGenerator help) throws StfException {
 		help.outputSection("MixedLoadTest runs workloads of math,mauve,nio,lang,concurrent unit tests");
 		help.outputText("");
 	}
 
 	public void pluginInit(StfCoreExtension stf) throws StfException {
+		// When used as a time based test, this STF test plugin accepts the following two parameters:
+		//   (1) timeLimit     	     : Time duration (e.g. 5s, 5m, 5h) for which the load should be run. 
+		//   (2) testCountMultiplier : Optional multiplier value to increase the workload externally.
+		StfTestArguments testArgs = stf.env().getTestProperties("timeLimit=[0m]","testCountMultiplier=[1]");
+		timeLimit = testArgs.get("timeLimit");
+		testCountMultiplier = Integer.parseInt(testArgs.get("testCountMultiplier"));
+		
+		// When We are using MixedLoadTest for default cases;. e.g. not in a time based load test scenario 
+		if ( timeLimit.equals("0m") && testCountMultiplier == 1) {
```

Comment:
Since you are using the default value to test whether timlimit was specified, the default value could be more descriptive (`unspecified` or `notset` or something?).

Reply:
Done. Test: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/5490/console